internal-api/internal-api-8/src/main/java/datadog/trace/util/stacktrace/DefaultStackWalker.java

```diff
@@ -0,0 +1,20 @@
+package datadog.trace.util.stacktrace;
+
+import java.util.Arrays;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+public class DefaultStackWalker extends AbstractStackWalker {
+
+  DefaultStackWalker() {}
+
+  @Override
+  public boolean isEnabled() {
+    return true;
+  }
+
+  @Override
+  <T> T doGetStack(final Function<Stream<StackTraceElement>, T> consumer) {
+    return consumer.apply(Arrays.stream(new Throwable().getStackTrace()));
```

Comment:
This would avoid allocation of a throwable object unnecessarily 
```suggestion
    return consumer.apply(Arrays.stream(Thread.currentThread().getStackTrace()));
```